hadoop-project/pom.xml

```diff
@@ -185,8 +185,8 @@
       --add-opens=java.base/sun.security.x509=ALL-UNNAMED
     </extraJavaTestArgs>
     <!-- Plugin versions and config -->
-    <maven-surefire-plugin.argLine>-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError ${extraJavaTestArgs}</maven-surefire-plugin.argLine>
-    <maven-surefire-plugin.version>3.0.0-M1</maven-surefire-plugin.version>
+    <maven-surefire-plugin.argLine>-Xmx2048m -Xss2m -XX:+HeapDumpOnOutOfMemoryError ${extraJavaTestArgs}</maven-surefire-plugin.argLine>
```

Comment:
Does the default stack size not work?

Reply:
This change was made because the `ResourceManager` module in #7335 reported an `unable to create new native thread` exception. Additionally, the third build result of #7335 appeared abnormal, and I suspected that the issue might be caused by the large number of compiled modules. I increased the `-Xss` size, which seemed to have some effect. The default value for this parameter is 1m.

The build report is as follows:
https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7335/3/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt

If either #7343 or #7335 is merged, the other PR can remove this change.